### ISSUE TITLE
Unusual Behavior from plaintext Attribute

### ISSUE THREAD

**Issue by bsmcgillis** [voku/simple_html_dom#18]
When I scrape a review on Yelp and get its plaintext, I get long character-replacing strings like this "It____SIMPLE_HTML_DOM__VOKU__AMP____#39;s obvious having grown up in the community that Dave____SIMPLE_HTML_DOM__VOKU__AMP____#39;s is here to stay."

This appears to be coming from the HtmlDomParser class. Is there something I can do to avoid this string replacement? I would prefer to just have the HTML entities themselves, like `&#39;`. That way, I could just use htmlspecialchars_decode to replace them. 

I tried innerHtml and innerText, but they both left a lot of HTML tags, which is problematic. 

In the meantime, I'm just reversing the process done by HtmlDomParser and then calling htmlspecialchars_decode.

**Comment by voku** [comment]
Hi, can you please show me your code example, so that I can create a PHPUnit test case and a fix or a other solution for you.

**Comment by bsmcgillis** [comment]
Here is a script that should return a good sample: 

`$review_page = 'https://www.yelp.com/biz/daves-auto-center-centerville';

		
		$dom = HtmlDomParser::file_get_html( $review_page );
		if( ! $dom ){
			echo 'No dom' . "\n"; //@DEBUG
			 return;
		}

		$reviews_container = $dom->find('.review-list', 0);
    	if( ! $reviews_container || ! $reviews_container->children(1) ){
    		echo 'No review container for ' . $review_account->business->name . "\n";
    		return;
    	}
    	$reviews_list = $reviews_container->children(1)->children();

		foreach( $reviews_list as $review ){

    		$review_content = $review->find('.review-content', 0)->find('p', 0);
    		if( ! $review_content ) continue;

    		$review_text_raw = $review_content->plaintext;
    		
    		echo $review_text_raw . "\n";
    		break;
    	}`

**Comment by voku** [comment]
Thanks for the bug-report and for the code-example. Fixed in version "4.1.5".

PS: you can use the find command "like" a jQuery selector, e.g. ...

```php
$review_page = "https://www.yelp.com/biz/daves-auto-center-centerville";	
$dom = HtmlDomParser::file_get_html( $review_page );

$reviews = [];
foreach ($dom->find('.review-content p') as $review) {
  $reviews[] = $review->plaintext . "\n";
}

var_dump($allReviews); 
```

**Comment by voku** [comment]
fixed via https://github.com/voku/simple_html_dom/commit/b8fd3094da6be3a5fda2a17642f12b81bc24cb71#diff-1e163f80dacdd100af6446e4282fa7caR733
